src/pytypes/onefuzztypes/models.py

```diff
@@ -494,20 +494,43 @@ class WorkerDoneEvent(BaseModel):
     stdout: str
 
 
-class WorkerEvent(BaseModel):
-    event: Union[WorkerDoneEvent, WorkerRunningEvent]
+class WorkerEvent(EnumModel):
+    done: Optional[WorkerDoneEvent]
+    running: Optional[WorkerRunningEvent]
+
+
+class SettingUpEventData(BaseModel):
+    tasks: List[UUID]
 
 
 class NodeStateUpdate(BaseModel):
     state: NodeState
+    data: Optional[SettingUpEventData]
+
+    @validator("data")
+    def check_data(cls, data: Optional[SettingUpEventData], values: Any) -> Optional[SettingUpEventData]:
+        if data:
+            state = values.get("state")
+            if state and state != NodeState.setting_up:
+                raise ValueError("data for node state update event does not match state = %s" % state)
+
+        data
```

Comment:
Need a `return`.

Reply:
Fix all the state.(Erendira-_$ all needs\%._ofcrditnow.